src/sqlfluff/core/dialects/dialect_bigquery.py

```diff
@@ -317,3 +318,11 @@ class TypelessStructSegment(BaseSegment):
             optional=True,
         ),
     )
+
+
+bigquery_dialect.replace(
```

Comment:
With this code, I would expect the change to `dialect_ansi.py` would be unnecessary, but `sqlfluff parse` of the test query shows that without the ANSI change, `month` is parsed as a column reference. 🤷‍♂️